.github/workflows/release.yml

```
@@ -0,0 +1,73 @@
+name: Release with SLSA provenance
+
+on:
+  push:
+    tags:
+      - 'v*'
+    branches:
+      - 'release/**'
+
+permissions:
+  contents: read
+  packages: write
+  id-token: write # Required for SLSA provenance generation
+
+jobs:
+  build-and-provenance:
+    strategy:
+      matrix:
+        platform: [linux/amd64, linux/arm64]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+        
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          
+      - name: Extract metadata for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ghcr.io/${{ github.repository }}
+          tags: |
+            type=semver,pattern={{version}}
+            type=sha,format=long
+            
+      - name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          platforms: ${{ matrix.platform }}
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          
+      - name: Generate SLSA provenance
+        uses: slsa-framework/slsa-github-generator/container@v1.9.0
+        with:
+          image: ghcr.io/${{ github.repository }}
```

from SLSA_Security_Expert
SLSAプロベナンス生成時に、registry-tokenを使用して認証を行うべきです。また、固定のSHAハッシュを使用すべきです。

```suggestion
      - name: Generate SLSA provenance
        uses: slsa-framework/slsa-github-generator/container@c2cdcb1c6d56edb669c5c5dd00648ec9f0372fa9 # v1.9.0
        with:
          image: ghcr.io/${{ github.repository }}
          digest: ${{ steps.build.outputs.digest }}
          registry-username: ${{ github.actor }}
          registry-password: ${{ secrets.GITHUB_TOKEN }}
```